File: A4.Empower-Git/A4.Empower/A4.BAL/SalesMarketing/StatusViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace A4.BAL
{
    public class StatusViewModel
    {
        public StatusViewModel()
        {
            DailyCallModel = new List<DailyCallModel>();
            DdlSaleStatus = new List<DropDownList>();
            StatusCompanyModel = new StatusCompanyModel();
        }
        public StatusCompanyModel StatusCompanyModel { get; set; }
        public List<DailyCallModel> DailyCallModel { get; set; }
        public List<DropDownList> DdlSaleStatus { get; set; }
    }

    public class StatusCompanyModel
    {
        public string CompanyId { get; set; }
        public string CompanyName { get; set; }
        public string CompanyAddress { get; set; }
        public string EmailId { get; set; }
        public string CompanyTelePhoneNo { get; set; }

    }

    public class DailyCallModel
    {
        public string Id { get; set; }

        public string Name { get; set; }

        [Required(ErrorMessage = "Please enter date & Time")]
        public DateTime CallDateTime { get; set; }

        [Required(ErrorMessage = "Please enter Description")]
        public string Description { get; set; }

        public string SalesStatusId { get; set; }

        public string SalesCompanyId { get; set; }
    }
}

File: ClientApp/src/app/models/sales-marketing/Status-view.model.ts
import { DropDownList } from "../common/dropdown";

export class StatusModel {
    constructor(dailyCallModel?: DailyCall[], totalCount?: number, ddlSaleStatus?) {
        this.dailyCallModel = new Array<DailyCall>();
        this.totalCount = totalCount;
        this.statusCompanyModel=new StatusCompanyModel();
    }
    public dailyCallModel: DailyCall[];
    public listStatus: StatusCompany[];
    public ddlSaleStatus: DropDownList[];
    public statusCompanyModel: StatusCompanyModel;
    public totalCount: number;
}
export class StatusCompanyModel {
    public companyId: string;
    public companyName: string;
    public companyAddress: string;
    public companyTelePhoneNo: string;
    public emailId:string;

}
export class StatusCompany {
    public id: string;
    public salesCompanyId: string;
    public callDateTime: string;
    public salesStatusId:string;
    public description: string;

}

export class DailyCall {
    public id: string;
    public callDateTime: Date;
    public description: string;
    public salesCompanyId: string;
}